postproengine/wake_meandering.py

```diff
@@ -0,0 +1,365 @@
+# Get the location where this script is being run
+import sys, os
+scriptpath = os.path.dirname(os.path.realpath(__file__))
+basepath   = os.path.dirname(scriptpath)
+# Add any possible locations of amr-wind-frontend here
+amrwindfedirs = ['../',
+                 basepath]
+for x in amrwindfedirs: sys.path.insert(1, x)
+
+from postproengine import registerplugin, mergedicts, registeraction
+import postproamrwindsample_xarray as ppsamplexr
+import postproamrwindsample as ppsample
+import numpy as np
+import pandas as pd
+import os
+import pickle
+import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+# samwichdir = '/projects/wind_uq/gyalla/src/waketracking/'
+# sys.path.insert(0,samwichdir)
+try:
+    from samwich.dataloaders import PlanarData
+    from samwich.waketrackers import track
+    usesamwhich=True
+except:
+    usesamwhich=False
+
+"""
+Plugin for computing wake meandering statistics
+
+See README.md for details on the structure of classes here
+"""
+
+def get_wake_centers(u,YY,ZZ,yhub,method='ConstantArea',weighting=lambda u: np.ones_like(u),args=None):
+    datadict = {}
+    datadict['y'] = np.copy(YY[:,:].T)
+    datadict['z'] = np.copy(ZZ[:,:].T)
+    datadict['u'] = np.array([np.copy(u[i,:,:,0].T) for i in range(0,u.shape[0])])
```

Comment:
could do `u[:,:,:,0].transpose(0,2,1).copy()` instead of looping over the first index and using `.T`